docker-compose.yml

```diff
@@ -2,6 +2,16 @@ version: "3.1"
 services:
   shock:
     image: kbase/kb_shock:${TRAVIS_BRANCH:-latest}
+    command:
+      - "-template"
+      - "/kb/deployment/conf/.templates/shock-server.cfg.templ:/kb/deployment/conf/shock-server.cfg"
+      - "-wait"
+      - "tcp://mongo:27017"
```

Comment:
Is this where dockerize waits for the target to be up before continuing?

Reply:
Yes, also look at the next "-wait" parameter that checks to make sure that the mongoinit container completed loading into mongodb before trying to spin up Shock. Before there was a race condition between mongoinit and Shock as to whether mongorestore would complete before Shock tried to use it.

Reply:
That's pretty nifty.  What's on port 8080 in the mongoinit container?